### PR TITLE
Add a short wait for the login fieldset to open

### PR DESCRIPTION
**Pull request by jbickar**
# READY FOR REVIEW

# Summary
- I've been getting intermittent errors like this, and I think this change will resolve them:

```
    And I am logged in as a user with the "administrator" role # SWSDrupalContext::assertAuthenticatedByRole()
      element not visible
        (Session info: chrome=56.0.2924.87)
        (Driver info: chromedriver=2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6),platform=Mac OS X 10.11.6 x86_64) (WARNING: The server did not provide any stacktrace information)
      Command duration or timeout: 51 milliseconds
      Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
      System info: host: 'TTS-MBP2011.stanford.edu', ip: '171.64.25.111', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.11.6', java.version: '1.8.0_73'
      Driver info: org.openqa.selenium.chrome.ChromeDriver
      Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6), userDataDir=/var/folders/h3/ghpj682n60984x76h550ynjw0000gp/T/.org.chromium.Chromium.HjeINJ}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=56.0.2924.87, platform=MAC, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
      Session ID: f26f26113b2a4503070ded54316af868 (WebDriver\Exception\ElementNotVisible)
```

This fix is mo' betta than #140, and I've been testing this one for a few days.

# Needed By (Date)
- No deadline

# Urgency
- Not killer. I can test on this branch for a while and see if it resolves my issue.

# Steps to Test

1. Run any one Scenario with a "Given I am logged in as a user with the 'role' role" step _with_ `@javascript`
2. Then run any one Scenario with the same step _without_ `@javascript`.

# Affected Projects or Products
- *ARRYTHING*

# Associated Issues and/or People
- Anyone who should be notified? @kbrownell

### CODE REVIEW

**Comment by jbickar** [comment]
@sherakama FYI

File: includes/bootstrap/SWSDrupalContext.php [hunk 1]
```diff
@@ -75,11 +75,15 @@ public function login() {
...
     $localuserlogin = $element->findLink("Local User Login");
+    // If we can find a link, that means we're using Javascript.
     if (!is_null($localuserlogin)) {
       // click on the Local User Login link to expose the user name and password fields
       // See Behat\Mink\Element\TraversableElement::clickLink
       $element->clickLink("Local User Login");
+      // Since we're using Javascript, we can use wait().
+      $this->getSession()->wait(5000, '(typeof(jQuery)=="undefined" || (0 === jQuery.active && 0 === jQuery(\':animated\').length))');
     }
+    // Shamelessly stolen from iWaitForAjaxToFinish().
```
Comment (sherakama) [inline includes/bootstrap/SWSDrupalContext.php]: This comment can go but this looks more like the code we are looking for. 
I wonder if we have a check that is more concrete than the $localuserlogin.

**Comment by sherakama** [comment]
@jbickar 

I changed this a bit more as I wanted a more concrete way of determining if the session is currently using javascript or not. Please review.

File: includes/bootstrap/SWSDrupalContext.php [hunk 2]
```diff
@@ -70,24 +70,28 @@ public function login() {
...
-    // find the Local User Login link - it's only findable in the browser, with Javascript
-    // See Behat\Mink\Element\TraversableElement::findLink
-    $localuserlogin = $element->findLink("Local User Login");
-    // If we can find a link, that means we're using Javascript.
-    if (!is_null($localuserlogin)) {
+
+    // @todo, find out if there is a more concrete way of if JS is enabled.
+    $session = $this->getSession();
+    $element = $session->getPage();
+    $javascript_enabled = method_exists($session, 'executeScript');
```
Comment (sherakama) [inline includes/bootstrap/SWSDrupalContext.php:77]: Here is where I am checking for the existence of executeScript method that is only available to javascript enabled session drivers.

Comment (sherakama) [reply includes/bootstrap/SWSDrupalContext.php:77]: The other option would be to check for the selenium instance of the driver but it is possible they we swap selenium for some other sort of JS enabled driver in the future.

**Comment by sherakama** [comment]
@jbickar ping

**Comment by jbickar** [comment]
https://www.youtube.com/watch?v=arCITMfxvEc

**Comment by pookmish** [comment]
For me this only seems to work when webauth is enabled AND using javascript.

**Comment by jbickar** [comment]
I think this also might break on Anchorage/SAML sites. Need to test but getting `git-crypt` errors with this branch currently.

**Comment by jbickar** [comment]
Haven't forgotten this, but haven't gotten back to testing it yet either.

**Pull request closed without merging**